### PR TITLE
Fix issues with OpenShift sidecar handling

### LINKED ISSUE #11883: Incorrect reference to "quarkus.openshift.sidecar" in the "Deploy to Kubernetes" guide.
**Description**
The guide for "Deploying to Kubernetes" contains the list of accepted configuration entries for OpenShift in https://quarkus.io/guides/deploying-to-kubernetes#openshift.
For creating a sidecar, the documentation instructs to use "quarkus.openshift.sidecars", but that entry does not work. 
After support from the core team, we found that the right entry should be "quarkus.openshift.containers".

Note that this somehow differs from the Kubernetes extension, that uses "quarkus.kubernetes.sidecars" for creating sidecars.

**Implementation ideas**
Just update the documentation to match actual entries.
Or even better, link to the "all-config" page with a pre-populated filter to "quarkus.openshift" or "quarkus.kubernetes" as needed.

### PR DESCRIPTION
**Pull request by geoand**
Based on discussions at: https://quarkusio.zulipchat.com/#narrow/stream/187030-users/topic/Openshift.2FKubernetes.20extension.20.2B.20sidecar

Fixes: #11883

### CODE REVIEW

File: extensions/kubernetes/vanilla/deployment/src/main/java/io/quarkus/kubernetes/deployment/OpenshiftConfig.java [hunk 1]
```diff
@@ -325,7 +334,16 @@ public ProbeConfig getReadinessProbe() {
     }
 
     public Map<String, ContainerConfig> getSidecars() {
-        return containers;
+        if (!containers.isEmpty() && !sidecars.isEmpty()) {
+            // done in order to make migration to the new property straight-forward
+            throw new IllegalStateException(
+                    "'quarkus.openshift.sidecars', 'quarkus.openshift.containers' cannot be used together. Please use the latter as the former has been deprecated");
```
Comment (gsmet) [inline extensions/kubernetes/vanilla/deployment/src/main/java/io/quarkus/kubernetes/deployment/OpenshiftConfig.java:340]:
```suggestion
                    "'quarkus.openshift.sidecars' and 'quarkus.openshift.containers' cannot be used together. Please use the former as the latter has been deprecated");
```
?

**Review by gsmet — DISMISSED** [review]

File: extensions/kubernetes/vanilla/deployment/src/main/java/io/quarkus/kubernetes/deployment/OpenshiftConfig.java (hunk 1 above)
Comment (geoand) [reply extensions/kubernetes/vanilla/deployment/src/main/java/io/quarkus/kubernetes/deployment/OpenshiftConfig.java]: Good catch!

**Comment by geoand** [comment]
@iocanel can you please review this? If it's good (or if it needs small changes), it would be good to make the cut for 1.8.0 and also be backported into the 1.7.x branch.

File: extensions/kubernetes/vanilla/deployment/src/main/java/io/quarkus/kubernetes/deployment/KubernetesProcessor.java [hunk 2]
```diff
@@ -723,13 +724,13 @@ private void applyBuildItems(Session session,
...
 
         kubernetesEnvs.forEach(e -> {
-            session.resources().decorate(e.getTarget(), new AddEnvVarDecorator(new EnvBuilder()
-                    .withName(EnvConverter.convertName(e.getName()))
-                    .withValue(e.getValue())
-                    .withSecret(e.getSecret())
-                    .withConfigmap(e.getConfigMap())
-                    .withField(e.getField())
-                    .build()));
+            String containerName = kubernetesName;
```
Comment (iocanel) [inline extensions/kubernetes/vanilla/deployment/src/main/java/io/quarkus/kubernetes/deployment/KubernetesProcessor.java:727]: Why do we have a different container name per deployment target?

**Review by iocanel — CHANGES_REQUESTED** [review]
Overall, it looks good to me.
There is a part where I'd like a clarification and maybe a comment on the source.
Also, a nice to have would be an update to the docs.

File: extensions/kubernetes/vanilla/deployment/src/main/java/io/quarkus/kubernetes/deployment/KubernetesProcessor.java (hunk 2 above)
Comment (geoand) [reply extensions/kubernetes/vanilla/deployment/src/main/java/io/quarkus/kubernetes/deployment/KubernetesProcessor.java:727]: We don't really, but what would we use instead to determine the container name?

Comment (iocanel) [reply extensions/kubernetes/vanilla/deployment/src/main/java/io/quarkus/kubernetes/deployment/KubernetesProcessor.java:727]: I see, we are just covering the edge case where the user decided to use different names. I am covered then!

**Review by iocanel — APPROVED** [review]
LGMT

**Pull request merged**